client/src/unifycr-sysio.c

```diff
@@ -463,12 +467,17 @@ int UNIFYCR_WRAP(__fxstat)(int vers, int fd, struct stat *buf)
 {
     int ret = 0;
 
+    DEBUG("fxstat was called for fd %d....\n", fd);
+
     /* check whether we should intercept this file descriptor */
     if (unifycr_intercept_fd(&fd)) {
+#if 0
+        /* TODO: do we really care? */
         if (vers != 3) {
             errno = EINVAL;
             return -1;
         }
+#endif
```

Comment:
@sandrain I'm not sure the vers check matters either... I wonder can we just delete this unless we can think of a reason it needs to be there?

Reply:
I have fixed this version check. It seems like `vers` should be `_STAT_VER`, according to `bits/stat.h`:

```
/* Versions of the `struct stat' data structure.  */                                                                                                                                                                                                                                                                          
#ifndef __x86_64__                                                                                                                                                                                                                                                                                                            
# define _STAT_VER_LINUX_OLD    1                                                                                                                                                                                                                                                                                             
# define _STAT_VER_KERNEL       1                                                                                                                                                                                                                                                                                             
# define _STAT_VER_SVR4         2                                                                                                                                                                                                                                                                                             
# define _STAT_VER_LINUX        3                                                                                                                                                                                                                                                                                             
                                                                                                                                                                                                                                                                                                                              
/* i386 versions of the `xmknod' interface.  */                                                                                                                                                                                                                                                                               
# define _MKNOD_VER_LINUX       1                                                                                                                                                                                                                                                                                             
# define _MKNOD_VER_SVR4        2                                                                                                                                                                                                                                                                                             
# define _MKNOD_VER             _MKNOD_VER_LINUX /* The bits defined below.  */                                                                                                                                                                                                                                               
#else                                                                                                                                                                                                                                                                                                                         
# define _STAT_VER_KERNEL       0                                                                                                                                                                                                                                                                                             
# define _STAT_VER_LINUX        1                                                                                                                                                                                                                                                                                             
                                                                                                                                                                                                                                                                                                                              
/* x86-64 versions of the `xmknod' interface.  */                                                                                                                                                                                                                                                                             
# define _MKNOD_VER_LINUX       0                                                                                                                                                                                                                                                                                             
#endif                                                                                                                                                                                                                                                                                                                        
                                                                                                                                                                                                                                                                                                                              
#define _STAT_VER               _STAT_VER_LINUX
```

Currently returns `EINVAL` if `vers` is not `_STAT_VER` (potentially could return `ENOSYS`).